File: solutions/java/darts/src/main/java/Darts.java
import java.lang.Math;
import java.util.List;

class Rule {

  public double outerEdge;
  public int points;

  public Rule(double outerEdge, int points) {
    this.outerEdge = outerEdge;
    this.points = points;
  }
}

class Darts {

  private static final List<Rule> rules = List.of(
    new Rule(1.0, 10),
    new Rule(5.0, 5),
    new Rule(10.0, 1)
  );

  int score(double xOfDart, double yOfDart) {
    final double dist = Math.hypot(xOfDart, yOfDart);
    for (Rule rule : rules) {
      if (dist <= rule.outerEdge) {
        return rule.points;
      }
    }
    return 0;
  }
}

File: solutions/csharp/darts/Darts.cs
public static class Darts
{
    private static readonly List<(Func<double, bool> Rule, int Score)> Rules =
    [
        (dist => dist <= 1,  10),
        (dist => dist <= 5,  5),
        (dist => dist <= 10, 1),
    ];
    
    public static int Score(double x, double y)
    {
        var dist = Math.Sqrt(x * x + y * y);;
        return Rules
            .FirstOrDefault(r => r.Rule(dist))
            .Score;
    }
}
